src/foam/nanos/pm/FoldReducePMLogger.java

```diff
@@ -129,9 +167,14 @@ public DAO getDelegate() {
       }
 
       public void removeAll_(X x, long skip, long limit, Comparator order, Predicate predicate) {
-        resetState();
+        if ( predicate == null && skip == 0 && limit == MAX_SAFE_INTEGER ) {
+          // Remove all
+          resetState();
```

Comment:
resetState() should also clear the captureEnabledList.

Reply:
is it possible that you would want to clear pms (time, min, max) but still keep capture flag when running benchmark runner?